src/tiledimage.js

```diff
@@ -2223,6 +2226,8 @@ $.extend($.TiledImage.prototype, $.EventSource.prototype, /** @lends OpenSeadrag
         tile.tiledImage = this; //unloaded with tile.unload(), so we need to set it back
         // does nothing if tile.cacheKey already present
 
+        $.console.assert(dataType !== undefined, "TileSource::downloadTileStart must return a dataType.");
```

Comment:
Even if I wanted to allow attempt to automatically deduct data type (if not provided), the usage of `addCache` below with callback requires the argument and fails otherwise, so it is safer to explicitly test.

Reply:
This just logs an error… is there anything else we should do, like abort the action?

Reply:
Not necessarily. The main idea was to give clear error cause, it would fail elsewhere with not-so-clear message. If someone implements custom data getter, they will get errors if they don't specify the format they return to the system. It's up to them to fix, I do not expect this to be issue elsehwere than in dev set-ups.